<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE chapter
[
<!ENTITY % entities SYSTEM "../../shared/syslog-ng-entities.ent">
%entities;
]>
<simplesect xml:id="simplesect-tid" xmlns="http://docbook.org/ns/docbook">
    <title>TID is already used</title>
    <indexterm>
        <primary>troubleshooting</primary>
        <secondary>TID is already used</secondary>
    </indexterm>
    <informaltable frame="topbot" colsep="0" rowsep="0">
        <tgroup cols="2">
            <colspec colnum="1" colwidth="40pt"/>
            <tbody>
                <row>
                    <entry>Error message:
                    </entry>
                    <entry>
                        <synopsis>TID is already used; proto='0x202c6c0', 
TID='61b6456d2f02052780d0d8930cbd043857c2463fcb6014b748b1450595a682', 
client='10.140.35.9'
Syslog connection closed;</synopsis>
                    </entry>
                </row>
                <row>
                    <entry>Description: 
                    </entry>
                    <entry>
                        <para>When a client using Reliable Log Transfer Protocol™ (RLTP™) connects to the server for the first time, it generates a persistent ID and sends it to the server during the handshake process. This is the <emphasis role="bold">TID</emphasis>.</para>
                        <para>If the client loses the connection to the server silently, for example, the UTP cable is pulled from the host or other network issues happen, the server is unable to detect the connection loss.</para>
                        <para>If the client tries to reconnect within a short time interval, it will send the same TID. However, the server allows only one connection with the same TID. As the server “thinks” that it already has a live connection with this TID, it drops the new connection due to the duplicated TID.</para>
                    </entry>
                </row>
                <row>
                    <entry>Solution: 
                    </entry>
                    <entry>
                        <para>This error is eliminated automatically because the RLTP™ server will close the connection if there were no new messages from the client within the timeout frame. Once the timeout period of the RLTP™ server has passed, the client will be able to reconnect to the server (when the <parameter>time_reopen()</parameter> of the client has elapsed).</para>
                        <para>If this error message appears regularly, it means that your network may be unstable, and sometimes the client loses the connection to the server in an abnormal way.</para>
                    </entry>
                </row>
            </tbody>
        </tgroup>
    </informaltable>
</simplesect>
